python/redmonster/physics/zpicker2.py

```diff
@@ -138,11 +139,17 @@ def classify_obj(self, zfindobjs, zfitobjs, flags, rchi2threshold=0.005):
                             fiberminvecs.append(
                                     zfitobjs[itemp].minvectors[ifiber][imin])
                         except IndexError as e:
-                            print "%r" % e
-                            fibermins.append( \
-                                    n.max(zfitobjs[itemp].chi2vals[ifiber]) / \
-                                    (self.dof[ifiber] - zfindobjs[itemp].npoly))
+                            #print "%r" % e
+                            #fibermins.append( \
+                            #        n.max(zfitobjs[itemp].chi2vals[ifiber]) / \
+                            #        (self.dof[ifiber] - zfindobjs[itemp].npoly))
+
+                            # this is ok, might be no solution
+                            #print "WARNING no z fitted for fiber #%d, class #%d, zid #%d"%(ifiber,itemp,imin)
+
+                            fibermins.append(100000.)
```

Comment:
Okay, so looking into this a bit further - it looks like it should be fixed by lines 171 and 188 in zfitter.py , unless there's some other case where an empty list is sneaking through.  I don't know if Julien was seeing this crash before or after I added this.

Regardless, maybe a better way to handle it would be to simply have a try/except block here - it'll try to add the maximum value (as in my version), and if a ValueError is raised, then fall back on the clunkier solution of just adding a large chi2 value.